File: FPG.Core/Algorithms/Salting/AdditiveSaltingAlgorithm.cs
using System;
using System.Collections.Generic;
using System.Text;

using FPG.Interfaces;

namespace FPG.Algorithms
{
    public class AdditiveSaltingAlgorithm : ISaltingAlgorithm
    {
        public String Salt(IEnumerable<String> inputs, IEnumerable<String> salts)
        {
            StringBuilder sb = new StringBuilder();
            IEnumerator<String> inputEnumerator = inputs.GetEnumerator();
            IEnumerator<String> saltEnumerator = salts.GetEnumerator();

            // Keyword
            inputEnumerator.MoveNext();
            sb.Append(inputEnumerator.Current);

            // Random Salt
            saltEnumerator.MoveNext();
            sb.Append(saltEnumerator.Current);
            // User Salt
            saltEnumerator.MoveNext();
            sb.Append(saltEnumerator.Current);

            return sb.ToString();
        }
    }
}

File: lib/core/algorithms/salting/AdditiveSaltingAlgorithm.dart
import 'package:fpg_mobile/core/interfaces/ISaltingAlgorithm.dart';

class AdditiveSaltingAlgorithm implements ISaltingAlgorithm {
  @override
  String salt(List<String> inputs, List<String?> salts) {
    StringBuffer sb = StringBuffer();
    Iterator<String> inputEnumerator = inputs.iterator;
    Iterator<String?> saltEnumerator = salts.iterator;

    // Keyword
    inputEnumerator.moveNext();
    sb.write(inputEnumerator.current);

    // Random Salt
    saltEnumerator.moveNext();
    sb.write(saltEnumerator.current);
    // User Salt
    saltEnumerator.moveNext();
    sb.write(saltEnumerator.current);

    return sb.toString();
  }
}
